### PR TITLE
Robot ready function

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/IsRobotReady.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import edu.greenblitz.tobyDetermined.commands.swerve.MoveToGrid.Grid;
+import edu.greenblitz.tobyDetermined.subsystems.swerve.SwerveChassis;
+import edu.greenblitz.tobyDetermined.subsystems.telescopicArm.Elbow;
+import edu.greenblitz.tobyDetermined.subsystems.telescopicArm.Extender;
+
+public class IsRobotReady {
+	public static boolean isArmReady(){
+		switch (Grid.getInstance().getSelectedHeight()){
+			case HIGH:
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/IsRobotReady.java:12]: DOC

File: src/main/java/edu/greenblitz/tobyDetermined/IsRobotReady.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+						|| RobotMap.TelescopicArm.PresetPositions.CUBE_MID.angleInRadians - Elbow.getInstance().getAngle() <= RobotMap.TelescopicArm.Elbow.ANGLE_TOLERANCE
+						&& RobotMap.TelescopicArm.PresetPositions.CUBE_MID.distance -  Extender.getInstance().getLength() <= RobotMap.TelescopicArm.Extender.LENGTH_TOLERANCE;
+			case LOW:
+				return RobotMap.TelescopicArm.PresetPositions.LOW.angleInRadians - Elbow.getInstance().getAngle() <= RobotMap.TelescopicArm.Elbow.ANGLE_TOLERANCE
+						&& RobotMap.TelescopicArm.PresetPositions.LOW.distance -  Extender.getInstance().getLength() <= RobotMap.TelescopicArm.Extender.LENGTH_TOLERANCE;
+		}
+		return false;
+	}
+
+	public static boolean isAtGoal(){
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/IsRobotReady.java:29]: Not sure about the name , maybe "isatlocation?" 
and doc

File: src/main/java/edu/greenblitz/tobyDetermined/IsRobotReady.java (hunk 1 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/IsRobotReady.java:12]: code is as self explanatory as can be doc would sound the same

File: src/main/java/edu/greenblitz/tobyDetermined/Field.java [hunk 3]
```diff
@@ -74,6 +77,19 @@ public static Pose2d[] getLocationsOnRedSide(){
             return mirrorPositionsToOtherSide(locationsOnBlueSide);
         }
 
+        private static final HashSet<Integer> CUBE_INDICES= new HashSet(Arrays.asList(1,4,7));
+
+        public static boolean indexOfCube(int index){
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/Field.java:82]: isindexofcube

File: src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToGrid/Grid.java [hunk 4]
```diff
@@ -34,6 +35,16 @@ public void updateAlliance(){
         }
     }
 
+    @Deprecated
+    public void putGridToShuffleboard(){
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToGrid/Grid.java:39]: delete

File: src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToGrid/Grid.java [hunk 5]
```diff
@@ -64,19 +74,44 @@ public Pose2d getSelectedPosition() {
...
+    public RobotMap.TelescopicArm.PresetPositions getSelectedConePosition(){
+        switch (getSelectedHeight()){
+            case LOW:
+                return RobotMap.TelescopicArm.PresetPositions.LOW;
+            case MEDIUM:
+                return RobotMap.TelescopicArm.PresetPositions.CONE_MID;
+            case HIGH:
+                return RobotMap.TelescopicArm.PresetPositions.CONE_HIGH;
+        }
+        return null; //todo make not null </3
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToGrid/Grid.java:95]: make not null

File: src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToGrid/Grid.java [hunk 6]
```diff
@@ -64,19 +74,44 @@ public Pose2d getSelectedPosition() {
...
+    public RobotMap.TelescopicArm.PresetPositions getSelectedCubePosition(){
+        switch (getSelectedHeight()){
+            case LOW:
+                return RobotMap.TelescopicArm.PresetPositions.LOW;
+            case MEDIUM:
+                return RobotMap.TelescopicArm.PresetPositions.CUBE_MID;
+            case HIGH:
+                return RobotMap.TelescopicArm.PresetPositions.CUBE_HIGH;
+        }
+        return null; //todo make not null </3
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToGrid/Grid.java:107]: also

**Review by yonnbro — APPROVED** [review]

**Pull request merged**